internal/db/schema/migrations/postgres/9/01_managed_groups.up.sql

```diff
@@ -0,0 +1,27 @@
+begin;
+
+-- The base abstract table
+create table auth_managed_group (
+  public_id wt_public_id
+    primary key,
+  auth_method_id wt_public_id
+    not null,
+  -- Ensure that if the auth method is deleted (which will also happen if the
+  -- scope is deleted) this is deleted too
+  constraint auth_method_fkey
+    foreign key (auth_method_id) -- fk1
+      references auth_method(public_id)
+      on delete cascade
+      on update cascade
```

Comment:
There should be a unique constraint for auth_method_id + public_id which should be the foreign key used by the subtypes. ([credential_store example](https://github.com/hashicorp/boundary/blob/mgaffney-vault/internal/db/schema/migrations/postgres/10/02_credential.up.sql#L12-L16))

Reply:
I'm assuming the reason for this is so that the subtype can't link only to the public ID but with a different auth method ID?

Added.